R/Plot_dose_response.R

```diff
@@ -52,17 +67,22 @@ PlotDoseResponse <- function (data, save.file = FALSE, pair.index = NULL, Emin =
     # plot dose-response matrix
     axis.x.text <- round(as.numeric(colnames(response.mat)), 1)
     axis.y.text <- round(as.numeric(rownames(response.mat)), 1)
-    dose.response.p <- ggplot(data.plot, aes_string(x = "x", y = "y")) + geom_tile(aes_string(fill = 'Inhibition')) +
-      geom_text(aes_string(fill = 'Inhibition', label = 'Inhibition')) +
-      scale_fill_gradient2(low = "green", high = "red", midpoint = 0, name = "Inhibiton (%)") +
-      scale_x_discrete(labels = axis.x.text) + scale_y_discrete(labels = axis.y.text) +
-      xlab(paste(drug.col, unit.text, sep = " ")) + ylab(paste(drug.row, unit.text, sep = " "))
-    dose.response.p <- dose.response.p + theme(axis.text.x = element_text(color = "red", face = "bold", size = 15))
-    dose.response.p <- dose.response.p + theme(axis.text.y = element_text(color = "red", face = "bold", size = 15))
-    dose.response.p <- dose.response.p + theme(axis.title = element_text(size = 15))
-    dose.response.p <- dose.response.p + ggtitle(plot.title) + theme(plot.title = 
-                                                                    element_text(size = 25))
-    
+    dose.response.p <- ggplot2::ggplot(
```

Comment:
The NAMESPACE file did not seem to have that effect when I tried to build the project in RStudio.  However you're right in general that imports can be arranged so that 'ggplot2::' isn't needed in all these calls.

Speaking personally, I think the code is much more clear this way, with explicit namespace designation.

Consider that you had a comment in a previous version, "use kriging from SpatialExtremes".  It's easier to follow a program without such a comment that simply calls SpatialExtremes::kriging.  Also, some of the functions here, like 'element_text' or 'theme' are very generic, and quite likely to occur in other packages besides ggplot2.

The case in which I would use the namespace implicitly is the case in which "ggplot2::" appears scattered all over the file.  When it appears everywhere, it's no longer a useful signal to the reader.  However it turns out that only a very localized part of this file has such calls.  Making the namespace explicit here seems to visually distinguish that part of the file from others.  So it also seems  useful on that count.

In the end this is all a judgment call and you are the long-term maintainer.  So If you would like these changes reverted I will do that.  However I wanted to be sure you knew my reasoning.